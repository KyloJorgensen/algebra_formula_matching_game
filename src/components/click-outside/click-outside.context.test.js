import React, { Component } from 'react';
import { shallow, mount } from 'enzyme';
import { ClickOutsideContext } from '../click-outside/click-outside.context.js';

describe('Render ClickOutsideContext.Consumer', () => {
	class ComponentDidMount extends Component { 
		constructor(props) {
			super(props);
		}

		componentDidMount() {
			this.props.addClickOutsideEventListener('uniqueTag', this.props.mockFunction);
		}

		componentWillUnmount() {
			this.props.removeClickOutsideEventListener('uniqueTag');
		}

		render () {
			return (<button></button>)
		}
	}
	ComponentDidMount.defaultProps = {
		addClickOutsideEventListener: () => {},
		mockFunction: () => {},
	};

	class UnmountButton extends Component { 
		constructor(props) {
			super(props);
			this.state = {
				mount: true,
			};
			this.handleClick = event => {
				const _state = this.state;
				_state.mount = false;
				this.setState(_state);
			};
		}

		render () {
			return (
				<div>
					<button className="unmount-button" onClick={this.handleClick}></button>
					{this.state.mount ? this.props.children : ''}
				</div>
			);
		}
	}

	let wrapper;

	it('should match the snapshot', () => {
		let props = {};
		wrapper = mount(
			<div>
				<ClickOutsideContext.Consumer>
					{context => {
						props.addClickOutsideEventListener = context.addClickOutsideEventListener;
						props.removeClickOutsideEventListener = context.removeClickOutsideEventListener;
						return (
							<div><p></p></div>
						);
					}}
				</ClickOutsideContext.Consumer>
			</div>
		);
		expect(wrapper).toMatchSnapshot();
	});



	describe('if no ClickOutsideContext.Provider', () => {
		describe('addClickOutsideEventListener', () => {
			it('should thorw error', () => {
				let props = {}
				wrapper = mount(
					<div>
						<ClickOutsideContext.Consumer>
							{context => {
								props = {
									addClickOutsideEventListener: context.addClickOutsideEventListener,
								}
								return '';
							}}
						</ClickOutsideContext.Consumer>
					</div>
				);

				expect(props.addClickOutsideEventListener).toThrow('Error: Cannot addClickOutsideEventListener no ClickOutsideContext.Provider element in parents');
			});
		});

		describe('removeClickOutsideEventListener', () => {
			it('should thorw error', () => {
				let props = {}
				wrapper = mount(
					<div>
						<ClickOutsideContext.Consumer>
							{context => {
								props = {
									removeClickOutsideEventListener: context.removeClickOutsideEventListener,
								}
								return '';
							}}
						</ClickOutsideContext.Consumer>
					</div>
				);

				expect(props.removeClickOutsideEventListener).toThrow('Error: Cannot removeClickOutsideEventListener no ClickOutsideContext.Provider element in parents');
			});
		});
	});
});


