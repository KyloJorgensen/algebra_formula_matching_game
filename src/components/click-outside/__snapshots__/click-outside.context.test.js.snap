// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ClickOutsideContext.Consumer should match the snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <div>
    <Context.Consumer>
      [Function]
    </Context.Consumer>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": <div>
      <div>
        <p />
      </div>
    </div>,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Context.Consumer>
        [Function]
      </Context.Consumer>,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": <div>
          <p />
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p />,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <p />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": <div>
        <div>
          <p />
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Context.Consumer>
          [Function]
        </Context.Consumer>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div>
            <p />
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <p />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
